rabbit-source/src/main/java/org/springframework/cloud/stream/module/rabbit/source/RabbitSource.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.rabbit.source;
+
+import org.aopalliance.aop.Advice;
+
+import org.springframework.amqp.core.AcknowledgeMode;
+import org.springframework.amqp.rabbit.config.RetryInterceptorBuilder;
+import org.springframework.amqp.rabbit.connection.ConnectionFactory;
+import org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer;
+import org.springframework.amqp.rabbit.retry.RejectAndDontRequeueRecoverer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.amqp.RabbitProperties;
+import org.springframework.boot.autoconfigure.amqp.RabbitProperties.Listener;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.Bindings;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.context.annotation.Bean;
+import org.springframework.integration.amqp.inbound.AmqpInboundChannelAdapter;
+import org.springframework.integration.amqp.support.DefaultAmqpHeaderMapper;
+import org.springframework.retry.interceptor.RetryOperationsInterceptor;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * A source module that receives data from RabbitMQ.
+ *
+ * @author Gary Russell
+ */
+@EnableBinding(Source.class)
+@EnableConfigurationProperties(RabbitSourceProperties.class)
+public class RabbitSource {
+
+	@Autowired
+	@Bindings(RabbitSource.class)
+	private Source channels;
+
+	@Autowired
+	private RabbitProperties rabbitProperties;
+
+	@Autowired
+	private RabbitSourceProperties properties;
+
+	@Autowired(required = false)
+	private ConnectionFactory rabbitConnectionFactory;
```

Comment:
This is left over from when I was trying to use separate config from the boot config. Needs to be removed.